streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -163,17 +166,19 @@ public void register(StateStore store, boolean loggingEnabled, StateRestoreCallb
             }
         } while (partitionNotFound && System.currentTimeMillis() < startTime + waitTime);
 
-        if (partitionNotFound)
+        if (partitionNotFound) {
             throw new StreamsException(String.format("task [%s]  Store %s's change log (%s) does not contain partition %s", taskId, store.name(), topic, partition));
+        }
+
+        this.stores.put(store.name(), store);
```

Comment:
can we move this down such that it is the last line? Again, just making sure that the store has been initialized before we put it into the map.